nsxt/resource_nsxt_policy_ip_block_quota_test.go

```diff
@@ -0,0 +1,254 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtPolicyIpBlockQuota_importBasic(t *testing.T) {
+	name := getAccTestResourceName()
+	testResourceName := "nsxt_policy_ip_block_quota.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyIpBlockQuotaCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyIpBlockQuotaMinimalistic(false),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNsxtPolicyIpBlockQuotaExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy IpBlockQuota resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy IpBlockQuota resource ID not set in resources")
+		}
+
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("Policy IpBlockQuota %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtPolicyIpBlockQuotaCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_ip_block_quota" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("Policy IpBlockQuota %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyIpBlockQuotaTemplate(createFlow bool, withContext bool) string {
+	var attrMap map[string]string
+	var ipblock string
+	if createFlow {
+		attrMap = accTestPolicyIpBlockQuotaCreateAttributes
+		ipblock = "test-ipv4"
+	} else {
+		attrMap = accTestPolicyIpBlockQuotaUpdateAttributes
+		ipblock = "test-ipv6"
+	}
+	context := ""
+	if withContext {
+		context = testAccNsxtPolicyMultitenancyContext()
+	}
+	return fmt.Sprintf(`
+resource "nsxt_policy_ip_block" "test-ipv4" {
+  %s
+  display_name = "%s-ipv4"
+  cidr         = "20.20.20.0/24"
+}
+
+resource "nsxt_policy_ip_block" "test-ipv6" {
+  %s
+  display_name = "%s-ipv6"
+  cidr         = "fc7e:0011:da43::0/48"
+}
+
+resource "nsxt_policy_ip_block_quota" "test" {
+  %s
+  display_name = "%s"
+  description  = "%s"
+
+  quota {
+    ip_block_paths        = [nsxt_policy_ip_block.%s.path]
```

Comment:
It's interesting that the API allows us to change the IP blocks a quota is applied to.

Reply:
I'm not sure what the behavior is if quota is already enforced - I think it would still allow to change the quota, even to smaller number